src/passes/OptimizeInstructions.cpp

```diff
@@ -1936,6 +1936,29 @@ struct OptimizeInstructions
           }
         }
       }
+      // TODO: This should be handled in previous rule block
+      // as more general (x + 5 <=> 7)
+      //
+      // x + C <=> 0  =>  x <=> -C
```

Comment:
I don't think this is true for the reasons in https://github.com/WebAssembly/binaryen/pull/3303#issuecomment-725632036

While `==`, `!=` allow such things, for comparisons there is a problem with overflows. You can imagine that an overflow leads to a (nonlinear!) addition or subtraction of `2^32` (or `2^64`) potentially on just one of the sides. That messes up math like that.

For example, `0x8 + 0x80000000 < 0` is true (it's a large signed number on the left), but `0x8 < -0x80000000` is false (as the right overflows after negation, and is a large negative number).

Anyhow, let's focus on fixing the fuzz bug first?

Reply:
> For example, 0x8 + 0x80000000 < 0 is true (it's a large signed number on the left), but 0x8 < -0x80000000

`neg(0x80000000) == -0x80000000 == 0x80000000` due to overflow, so:
`0x8 + 0x80000000 < 0` will be `0x8 < 0x80000000`

Reply:
Actually `-0x80000000` is not possible in 32-bit. It cause to overflow which lead to `0x80000000` value